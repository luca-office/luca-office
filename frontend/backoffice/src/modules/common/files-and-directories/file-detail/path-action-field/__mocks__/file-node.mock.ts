import {FileNode} from "shared/components"
import {BinaryType, NodeType} from "shared/enums"

export const fileNodeMock: FileNode = {
  binaryFileId: "spfosdkfsd",
  binaryFileUrl: "url@url.de",
  binaryType: BinaryType.Image,
  id: "sfpoksdf-fposjfd",
  name: "test-file",
  parentId: "sfopksdf-fsdpfosdk",
  tags: [],
  type: NodeType.File
}
