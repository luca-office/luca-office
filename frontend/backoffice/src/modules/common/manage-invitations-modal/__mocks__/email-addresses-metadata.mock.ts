import {EmailAddressesMetadata} from "../models/email-addresses-metadata"

export const emailAddressesMetadataMock: EmailAddressesMetadata = {
  validAddresses: [
    "testmail@test.de",
    "cap@cap3.de",
    "testmail@test.de",
    "testmai2l@test.de",
    "cap3@cap3.de",
    "cap@cap3.de"
  ],
  uniqueAddresses: ["testmail@test.de", "cap@cap3.de", "testmai2l@test.de", "cap3@cap3.de"],
  hasInvalidAddress: false,
  duplicatedAddresses: ["testmail@test.de"],
  alreadyInvitedAddresses: ["cap@cap3.de"]
}
