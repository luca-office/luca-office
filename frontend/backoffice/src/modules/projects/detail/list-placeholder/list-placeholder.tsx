import {css} from "@emotion/react"
import * as React from "react"
import {Button, Text} from "shared/components"
import {IconName} from "shared/enums"
import {Flex, fontColorLight, spacing, spacingMedium, spacingTiny, TextSize} from "shared/styles"

export interface ListPlaceholderProps {
  readonly actionText: string
  readonly onClick: React.MouseEventHandler
  readonly text: string
  readonly title: string
  readonly actionIcon?: IconName
  readonly disabled?: boolean
}

export const ListPlaceholder: React.FunctionComponent<ListPlaceholderProps> = ({
  actionIcon,
  actionText,
  disabled = false,
  onClick,
  text,
  title
}) => {
  return (
    <div className="list-placeholder" css={[Flex.column, styles.wrapper]}>
      <Text size={TextSize.Medium}>{title}</Text>
      <Text size={TextSize.Medium} customStyles={styles.text}>
        {text}
      </Text>
      <Button
        icon={actionIcon || IconName.Add}
        onClick={onClick}
        disabled={disabled}
        customStyles={styles.button}
        className={"placeholder-button"}>
        {actionText}
      </Button>
    </div>
  )
}

const styles = {
  wrapper: css({
    padding: spacingMedium,
    alignItems: "center",
    alignContent: "center",
    flex: "0 0 50%"
  }),
  text: css({
    color: fontColorLight,
    margin: spacing(spacingTiny, 0, spacingMedium)
  }),
  button: css({
    width: "auto",
    padding: spacing(0, spacingMedium)
  })
}
