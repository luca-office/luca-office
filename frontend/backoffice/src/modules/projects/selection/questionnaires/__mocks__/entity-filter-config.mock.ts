import {EntityFilterOptionState, EntityFilterOptionType, EntitySortOption} from "../../../../../enums"
import {EntityFilterConfig} from "../../../../../models"

export const entityFilterConfigMock: EntityFilterConfig = {
  sortBy: EntitySortOption.BY_NAME,
  state: EntityFilterOptionState.SHOW_ALL,
  type: EntityFilterOptionType.ALL_ENTITIES,
  search: ""
}
