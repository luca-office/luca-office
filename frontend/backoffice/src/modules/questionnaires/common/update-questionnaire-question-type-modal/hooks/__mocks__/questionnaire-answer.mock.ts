import {QuestionnaireAnswer} from "shared/models"

export const questionnaireAnswerMock = (
  questionId: UUID = "18316dac-2355-4e45-a7f3-f339461a488f"
): QuestionnaireAnswer => ({
  __typename: "QuestionnaireAnswer",
  createdAt: new Date(2020, 10, 5).toISOString(),
  modifiedAt: new Date(2020, 10, 15).toISOString(),
  id: "1fce8289-71fa-4568-bccc-329dc4cc67eb",
  text: "answer",
  questionId,
  isCorrect: false,
  position: 1
})
