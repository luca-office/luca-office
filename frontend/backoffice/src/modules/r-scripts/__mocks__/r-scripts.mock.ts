import {RScript} from "shared/models"

export const rScriptMock: RScript = {
  __typename: "RScript",
  id: "18de306c-a8f8-4311-9a2f-ca5418a245bd",
  createdAt: "2020-08-03 15:16:28.501000 +00:00",
  title: "title",
  description: "description",
  modifiedAt: "2020-08-03 15:16:28.501000 +00:00",
  archivedAt: null,
  gitCommitHash: "sd",
  version: "1.2",
  isEditable: true
}

export const rScriptsMock: RScript[] = [
  rScriptMock,
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2452d"},
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2453d"},
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2454d"},
  {...rScriptMock, id: "18de306c-a8f8-4311-9a2f-ca5418a2455d"}
]
