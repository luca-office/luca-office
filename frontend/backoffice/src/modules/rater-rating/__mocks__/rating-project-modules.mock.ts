import {projectModulesMockWithQuestionnaire, surveyInvitationsMock} from "shared/graphql/__mocks__"
import {RatingProjectModule} from "../models"

export const ratingProjectModulesMock: RatingProjectModule[] = projectModulesMockWithQuestionnaire.map(
  (projectModule, index) => ({
    ...projectModule,
    ratingsByParticipants: surveyInvitationsMock.map(({id}, index) => ({
      participantId: id,
      isFullyRated: index % 2 === 0,
      isNotRatable: false
    })),
    participantCount: index + 1,
    ratedParticipantCount: index
  })
)
