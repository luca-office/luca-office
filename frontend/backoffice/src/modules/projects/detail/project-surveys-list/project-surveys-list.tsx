import {css} from "@emotion/react"
import * as React from "react"
import {TableContainer} from "shared/components"
import {IconName} from "shared/enums"
import {Survey} from "shared/models"
import {
  boxHeightMedium,
  CustomStyle,
  FontWeight,
  headerHeight,
  spacingLarge,
  spacingSmall,
  spacingTiny,
  TextSize
} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {getIdKeyProjectTable, getSurveyColumns} from "../../config/detail-tables"
import {ListPlaceholder} from "../list-placeholder/list-placeholder"

export interface ProjectSurveysListProps extends CustomStyle {
  readonly deleteSurvey: (id: UUID) => void
  readonly isTestSurvey: boolean
  readonly navigateToSurvey: (id: UUID) => void
  readonly navigateToSurveyCreation: (isTestSurvey?: boolean) => void
  readonly navigateToProjectModules: () => void
  readonly surveyCreationDisabledMissingMeta: boolean
  readonly surveyCreationDisabledModuleCheckFailed: boolean
  readonly surveys: Survey[]
  readonly isLoading: boolean
}

export const ProjectSurveysList: React.FunctionComponent<ProjectSurveysListProps> = ({
  deleteSurvey,
  isTestSurvey,
  surveys,
  isLoading,
  surveyCreationDisabledMissingMeta,
  surveyCreationDisabledModuleCheckFailed,
  navigateToSurvey,
  navigateToSurveyCreation,
  navigateToProjectModules
}) => {
  const {t} = useLucaTranslation()
  const surveyCreationDisabled = surveyCreationDisabledMissingMeta || surveyCreationDisabledModuleCheckFailed
  const surveyPlaceholderText = t(
    surveyCreationDisabled ? "projects__surveys_placeholder_text" : "projects__surveys_placeholder_text_enabled"
  )
  const testSurveyPlaceholderText = t(
    surveyCreationDisabled
      ? "projects__test_surveys_placeholder_text"
      : "projects__test_surveys_placeholder_text_enabled"
  )

  return (
    <TableContainer
      entities={surveys}
      columns={getSurveyColumns({
        t,
        openCreation: () => navigateToSurveyCreation(isTestSurvey),
        deleteSurvey,
        surveyCreationDisabled,
        isTestSurvey
      })}
      entityKey={getIdKeyProjectTable}
      customStyles={projectSurveyStyles.tableContent}
      customHeaderRowStyles={projectSurveyStyles.tableHeaderRow}
      customRowStyles={() => projectSurveyStyles.tableRow}
      onClick={(survey: Survey) => navigateToSurvey(survey.id)}
      showFooter={true}
      customPlaceholderStyles={projectSurveyStyles.tablePlaceholder}
      showLoadingIndicator={isLoading}
      customPlaceholder={
        <ListPlaceholder
          title={
            isTestSurvey ? t("projects__test_surveys_placeholder_title") : t("projects__surveys_placeholder_title")
          }
          text={isTestSurvey ? testSurveyPlaceholderText : surveyPlaceholderText}
          actionText={t(
            surveyCreationDisabledModuleCheckFailed && !surveyCreationDisabledMissingMeta
              ? "projects__surveys_modules"
              : isTestSurvey
              ? "projects__surveys_add_test_survey"
              : "projects__surveys_add"
          )}
          disabled={surveyCreationDisabledMissingMeta}
          onClick={
            surveyCreationDisabledModuleCheckFailed
              ? navigateToProjectModules
              : () => navigateToSurveyCreation(isTestSurvey)
          }
          actionIcon={
            surveyCreationDisabledModuleCheckFailed && !surveyCreationDisabledMissingMeta
              ? IconName.ArrowRight
              : undefined
          }
        />
      }
    />
  )
}

const Dimension = {
  cardLabel: spacingLarge + spacingTiny // height and margin of card label
}

export const projectSurveyStyles = {
  tableContent: css({
    height: "100%",
    minHeight: 208,
    width: "100%"
  }),
  cardWrapper: css({
    flex: "1 1 auto",
    height: `calc(100% - ${Dimension.cardLabel}px)`,
    cursor: "initial",
    display: "grid",
    gridAutoColumns: "1fr",
    gridAutoRows: "minmax(0, 1fr) minmax(min-content, max-content)"
  }),
  addButton: css({
    width: "auto"
  }),
  footer: css({
    justifyContent: "flex-end"
  }),
  tableHeaderRow: css({
    padding: spacingSmall,
    fontSize: TextSize.Medium,
    fontWeight: FontWeight.Bold
  }),
  tableRow: css({
    height: boxHeightMedium
  }),
  tableRowHighlighted: css({
    backgroundColor: "rgb(255, 248, 235)"
  }),
  tablePlaceholder: css({
    minHeight: "initial",
    height: `calc(100% - ${headerHeight + 2 * spacingSmall}px)`
  })
}
