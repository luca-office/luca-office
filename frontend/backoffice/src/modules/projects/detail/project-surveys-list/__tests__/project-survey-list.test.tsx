import {mount} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Button, Table} from "shared/components"
import {IconName} from "shared/enums"
import {surveysMock} from "shared/graphql/__mocks__"
import {ListPlaceholder} from "../../list-placeholder/list-placeholder"
import {ProjectSurveysList, ProjectSurveysListProps} from "../project-surveys-list"

const moduleSpy = jest.fn()
const surveySpy = jest.fn()
const defaultProps: ProjectSurveysListProps = {
  surveyCreationDisabledMissingMeta: false,
  surveyCreationDisabledModuleCheckFailed: false,
  surveys: surveysMock,
  navigateToProjectModules: moduleSpy,
  navigateToSurveyCreation: surveySpy,
  navigateToSurvey: jest.fn(),
  deleteSurvey: jest.fn,
  isLoading: false,
  isTestSurvey: false
}

const getComponent = (props?: Partial<ProjectSurveysListProps>) => (
  <ProjectSurveysList {...{...defaultProps, ...props}} />
)

describe("project-surveys-list", () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it("renders correctly", () => {
    const component = getComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correct structure", () => {
    const component = getComponent()
    const tree = mount(component)

    expect(tree.find(Table)).toHaveLength(1)
    expect(tree.find(Button)).toHaveLength(surveysMock.length + 1)
  })
  describe("renders correct structure (no data)", () => {
    it("renders placeholder for surveys", () => {
      const component = getComponent({surveys: []})
      const tree = mount(component)

      expect(tree.find(Table)).toHaveLength(1)
      expect(tree.find(Table).find(".entity-row")).toHaveLength(0)
      expect(tree.find(Table).find(Button)).toHaveLength(2)
      expect(tree.find(Table).find(ListPlaceholder)).toHaveLength(1)
    })
    it("renders placeholder for test surveys", () => {
      const component = getComponent({surveys: [], isTestSurvey: true})
      const tree = mount(component)

      expect(tree.find(Table)).toHaveLength(1)
      expect(tree.find(Table).find(".entity-row")).toHaveLength(0)
      expect(tree.find(Table).find(Button)).toHaveLength(2)
      expect(tree.find(Table).find(ListPlaceholder)).toHaveLength(1)

      const placeholderBtn = tree.find(".placeholder-button").first()
      expect(placeholderBtn.prop("disabled")).toBe(false)
      expect(placeholderBtn.prop("icon")).toBe(IconName.Add)
    })
  })

  describe("renders buttons", () => {
    it("renders create button", () => {
      const component = getComponent()
      const tree = mount(component)

      expect(tree.find(Button).first().prop("disabled")).toBe(false)
    })
    it("renders create button for test surveys", () => {
      const component = getComponent({isTestSurvey: true})
      const tree = mount(component)

      expect(tree.find(Button).first().prop("disabled")).toBe(false)
    })
    it("renders disabled button", () => {
      const component = getComponent({surveys: [], surveyCreationDisabledMissingMeta: true})
      const tree = mount(component)

      expect(tree.find(Button).first().prop("disabled")).toBe(true)
      expect(tree.find(".placeholder-button").first().prop("disabled")).toBe(true)
    })
    it("renders disabled button for test surveys", () => {
      const component = getComponent({surveys: [], surveyCreationDisabledMissingMeta: true, isTestSurvey: true})
      const tree = mount(component)

      expect(tree.find(Button).first().prop("disabled")).toBe(true)
      expect(tree.find(".placeholder-button").first().prop("disabled")).toBe(true)
    })
    it("renders module button", () => {
      const component = getComponent({surveys: [], surveyCreationDisabledModuleCheckFailed: true, isTestSurvey: true})
      const tree = mount(component)

      expect(tree.find(Button).first().prop("disabled")).toBe(true)
      const placeholderBtn = tree.find(".placeholder-button").first()
      expect(placeholderBtn.prop("disabled")).toBe(false)
      expect(placeholderBtn.prop("icon")).toBe(IconName.ArrowRight)
    })
    it("renders module button for test surveys", () => {
      const component = getComponent({surveys: [], surveyCreationDisabledModuleCheckFailed: true})
      const tree = mount(component)

      expect(tree.find(Button).first().prop("disabled")).toBe(true)
      const placeholderBtn = tree.find(".placeholder-button").first()
      expect(placeholderBtn.prop("disabled")).toBe(false)
      expect(placeholderBtn.prop("icon")).toBe(IconName.ArrowRight)
    })
  })
})
