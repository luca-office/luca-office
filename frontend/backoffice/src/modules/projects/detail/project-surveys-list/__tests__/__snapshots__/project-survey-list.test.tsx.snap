// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project-surveys-list renders correctly 1`] = `
[
  <div
    className="css-p8sq3i-Table"
  >
    <div
      className="css-oxiieb-Table"
    >
      <div
        className="css-fuywjr-Table"
      >
        projects__surveys_table_title
      </div>
      <div
        className="css-fuywjr-Table"
      >
        projects__surveys_table_condition
      </div>
      <div
        className="css-fuywjr-Table"
      >
        projects__surveys_table_participants
      </div>
      <div
        className="css-fuywjr-Table"
      >
        projects__surveys_table_status
      </div>
      <div
        className="css-51lhxd-Table"
      >
        <div
          className="css-0"
          id="tooltip_42"
        >
          <button
            className="css-1cywa3f-Button"
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="css-95j2hm-Button"
            >
              <div
                className="css-mj8vo1-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
            </div>
          </button>
        </div>
        <div
          className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
          role="tooltip"
          style={
            {
              "backgroundColor": "#2e3032",
              "zIndex": 1,
            }
          }
        >
          projects__surveys_add
          <div
            className="react-tooltip-arrow styles-module_arrow__K0L3T"
            style={{}}
          />
        </div>
      </div>
    </div>
    <div
      className="css-tqgzls-Table"
    >
      <div
        className="css-1ihrgr1-Table"
      >
        <div
          className="entity-row css-s3x2fw-Table"
          onClick={[Function]}
        >
          <div
            className="css-fuywjr-Table"
          >
            my test scenario
          </div>
          <div
            className="css-fuywjr-Table"
          >
            projects__survey_overlay_closed_participation
          </div>
          <div
            className="css-fuywjr-Table"
          >
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="css-17zbm7y-Text"
              >
                projects__surveys_table_participants_column
              </div>
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              projects__surveys_table_participants_column_tooltip
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
          </div>
          <div
            className="css-fuywjr-Table"
          >
            <div
              className="css-16l5eqp-Text"
            >
              <div
                className="css-cj5xss-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              dashboard__details_time_ended
            </div>
          </div>
          <div
            className="css-51lhxd-Table"
          >
            <div
              className="css-0"
            >
              <div
                className="css-0"
                id="tooltip_42"
              >
                <button
                  className="css-zazuei-Button"
                  disabled={true}
                  onClick={[Function]}
                >
                  <div
                    className="css-95j2hm-Button"
                  >
                    <div
                      className="css-op5egs-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </button>
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1ihrgr1-Table"
      >
        <div
          className="entity-row css-s3x2fw-Table"
          onClick={[Function]}
        >
          <div
            className="css-fuywjr-Table"
          >
            my test scenario
          </div>
          <div
            className="css-fuywjr-Table"
          >
            projects__survey_overlay_closed_participation
          </div>
          <div
            className="css-fuywjr-Table"
          >
            <div
              className="css-0"
              id="tooltip_42"
            >
              <div
                className="css-17zbm7y-Text"
              >
                projects__surveys_table_participants_column
              </div>
            </div>
            <div
              className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
              role="tooltip"
              style={
                {
                  "backgroundColor": "#2e3032",
                  "zIndex": 1,
                }
              }
            >
              projects__surveys_table_participants_column_tooltip
              <div
                className="react-tooltip-arrow styles-module_arrow__K0L3T"
                style={{}}
              />
            </div>
          </div>
          <div
            className="css-fuywjr-Table"
          >
            <div
              className="css-16l5eqp-Text"
            >
              <div
                className="css-cj5xss-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              dashboard__details_time_ended
            </div>
          </div>
          <div
            className="css-51lhxd-Table"
          >
            <div
              className="css-0"
            >
              <div
                className="css-0"
                id="tooltip_42"
              >
                <button
                  className="css-zazuei-Button"
                  disabled={true}
                  onClick={[Function]}
                >
                  <div
                    className="css-95j2hm-Button"
                  >
                    <div
                      className="css-op5egs-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </button>
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="card-footer css-jb8qoz-CardFooter"
  />,
]
`;
